.github/actions/run-ios-simulator/src/main.ts

```diff
@@ -12,61 +12,62 @@ async function run(): Promise<void> {
         const iphoneToSimulate = core.getInput("iphoneToSimulate", { required: false });
         const args = core.getInput("arguments", { required: false });
 
-        let runtimeId = "";
-        const options : exec.ExecOptions = {};
-        options.listeners = {
-            stdout: (data: Buffer) => {
-                runtimeId += data.toString();
-            },
-        };
-
-        await execCmd("xcrun simctl list runtimes", options);
+        let runtimeId = await execCmd("xcrun simctl list runtimes");
 
         // Sample output: iOS 14.5 (14.5 - 18E182) - com.apple.CoreSimulator.SimRuntime.iOS-14-5
-        // and we want to extract "iOS 14.5"
+        // and we want to extract "iOS 14.5" and "com.apple.CoreSimulator.SimRuntime.iOS-14-5"
         // If we want to allow launching watchOS/tvOS simulators, replace the 'iOS' with an 'os' argument
-        const matches = runtimeId.match(/(iOS \d{1,2}(.\d{1,2})?)/g);
-        if (matches && matches.length > 0) {
-            runtimeId = matches[0].replace(" ", "");
-            core.info(`runtimeId: ${runtimeId}`);
+        const matches =/(?<runtime1>iOS \d{1,2}(.\d{1,2})?).*(?<runtime2>com\.apple\.CoreSimulator\.SimRuntime\.iOS-[0-9.-]+)/g.exec(runtimeId);
+        if (!matches || matches?.length == 0 || matches?.groups || matches?.groups?.runtime1.length == 0 || matches?.groups?.runtime2.length == 0) {
```

Comment:
You don't need the question marks if you already checked that the value is not falsy. Additionally, you're entering the if clause if `matches.groups` is truthy, meaning that if you did get some groups  to match, you'll set the result the failed. Finally, `matches.groups.runtime1` being truthy is stronger than `matches.length` being non-zero. So the condition can probably be simplified to something like:

```suggestion
        if (!matches?.groups?.runtime1 || !matches?.groups?.runtime2) {
```